subtask_9.3/src/DataBuffer.java

```diff
@@ -0,0 +1,28 @@
+import java.util.concurrent.ArrayBlockingQueue;
+import java.util.concurrent.BlockingQueue;
+import java.util.concurrent.ThreadLocalRandom;
+
+public class DataBuffer {
+
+    private BlockingQueue<Integer> buffer;
+
+    public DataBuffer(int capacity) {
```

Comment:
Already much better, but what about buffer overflow and underflow? I don't see any thread control logic in this case